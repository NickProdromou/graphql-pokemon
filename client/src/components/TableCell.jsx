import React from 'react';
import PropTypes from 'prop-types';
import Styled, {css} from 'styled-components';

import { spacing, colours } from '../style/variables';
import { type } from '../style/mixins/index';

const TableCell = ({children, title, wrap}) => (
  <StyledCell wrap={wrap} title={title}>
    { children }
  </StyledCell>
)

TableCell.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.bool, 
  wrap: PropTypes.bool,
}

TableCell.defaultProps = {
  title: false,
  wrap: false,
}

const StyledCell = Styled.div`  
  flex: 1 0 306px;
  align-items: flex-start;
  padding: ${spacing.small.level3};
  border: 1px solid #eee;

  ${(props) => props && props.wrap && css`
    display: flex;
    flex-wrap: wrap;
  `}    

  ${(props) => props && props.title && css`
   ${type("ui")}
    background: ${colours.accentColor};
    color: ${colours.whiteText};
    text-transform: uppercase;
  `}    
`

export default TableCell;
