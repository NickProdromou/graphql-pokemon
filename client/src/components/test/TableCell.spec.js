import React from "react";
import TableCell from "../TableCell";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("TableCell component", () => {
  
  test("renders with children", () => {
    const component = renderer.create(
      <TableCell>
        <div>
          <p>This is a children</p>
        </div>
      </TableCell>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
