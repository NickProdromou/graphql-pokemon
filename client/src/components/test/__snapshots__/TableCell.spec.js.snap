// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell component renders with children 1`] = `
<div
  className="sc-bdVaJa lnMYig"
  title={false}
  wrap={false}
>
  <div>
    <p>
      This is a children
    </p>
  </div>
</div>
`;
